src/main/java/org/openrewrite/java/migrate/javax/AddDefaultConstructorToEntityClass.java

```diff
@@ -0,0 +1,90 @@
+import java.util.stream.Collectors;
+
+// TODO: possibly rename to EntityNoArgConstructor?
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class AddDefaultConstructorToEntityClass extends Recipe {
+
+    @Override
+    public String getDisplayName() {
+        return "@Entity objects with constructors must also have a default constructor";
+    }
+
+    @Override
+    public String getDescription() {
+        return "When a Java Persistence API (JPA) entity class has a constructor with arguments, the class must also " +
+               "have a default, no-argument constructor. The OpenJPA implementation automatically generates the " +
+               "no-argument constructor, but the EclipseLink implementation does not.";
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+        final JavaTemplate defaultConstructorTemplate = JavaTemplate.builder("public #{}(){}").build();
+        return Preconditions.check(
+                Preconditions.or(
+                        new UsesType<>("javax.persistence.Entity", true),
+                        new UsesType<>("javax.persistence.MappedSuperclass", true)
+                        ),
+                new JavaIsoVisitor<ExecutionContext>() {
+                    @Override
+                    public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration cd, ExecutionContext ctx) {
+                        // Exit if class not annotated with either @Entity or @MappedSuperclass
+                        if (FindAnnotations.find(cd, "javax.persistence.Entity").isEmpty()
+                        && FindAnnotations.find(cd, "javax.persistence.MappedSuperclass").isEmpty()) {
+                            return cd;
+                        }
+
+                        // Check if class has a default no-arg constructor
+                        boolean hasDefaultConstructor = cd.getBody().getStatements().stream()
+                                .filter(statement -> statement instanceof J.MethodDeclaration)
+                                .map(J.MethodDeclaration.class::cast)
+                                .filter(methodDeclaration -> methodDeclaration.getMethodType().getName().equals("<constructor>"))
+                                .anyMatch(constructor -> constructor.getParameters().get(0).getClass().equals(J.Empty.class));
+
+                        // Exit if class already has default no-arg constructor
+                        if (hasDefaultConstructor) {
+                            return cd;
+                        }
+
+                        // Add default constructor with empty body
+                        // For testing simplicity, adding as last statement in class body
+                        cd = cd.withBody(defaultConstructorTemplate.apply(
+                                new Cursor(getCursor(), cd.getBody()),
+                                cd.getBody().getCoordinates().lastStatement(),
+                                cd.getSimpleName()
```

Comment:
This should be generating the correct `public <classname>(){}` constructor, and it did at one point.
But it suddenly started making `public Template <classname>(){}`, and I hadn't made any changes to the recipe code.

Reply:
For reference, 
```
final JavaTemplate defaultConstructorTemplate = JavaTemplate.builder("public #{}(){}").build();
```

Reply:
Fixed by using `.contextSensitive()` on the JavaTemplate builder